src/wp-includes/rest-api/endpoints/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,549 @@
+ */
+class WP_REST_Block_Types_Controller extends WP_REST_Controller {
+
+	/**
+	 * Instance of WP_Block_Type_Registry.
+	 *
+	 * @since 5.5.0
+	 * @var WP_Block_Type_Registry
+	 */
+	protected $block_registry;
+
+	/**
+	 * Instance of WP_Block_Styles_Registry.
+	 *
+	 * @since 5.5.0
+	 * @var WP_Block_Styles_Registry
+	 */
+	protected $style_registry;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->namespace      = 'wp/v2';
+		$this->rest_base      = 'block-types';
+		$this->block_registry = WP_Block_Type_Registry::get_instance();
+		$this->style_registry = WP_Block_Styles_Registry::get_instance();
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<namespace>[a-zA-Z0-9_-]+)',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<namespace>[a-zA-Z0-9_-]+)/(?P<name>[a-zA-Z0-9_-]+)',
+			array(
+				'args'   => array(
+					'name'      => array(
+						'description' => __( 'Block name' ),
+						'type'        => 'string',
+					),
+					'namespace' => array(
+						'description' => __( 'Block namespace' ),
+						'type'        => 'string',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks whether a given request has permission to read post block types.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
```

Comment:
```suggestion
	public function get_items_permissions_check( $request ) {
```

I don't think this is necessary for Core.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/317/commits/30053466afbc62ad11b238ba515328bfaa59cfdf